src/main/java/staffconnect/model/person/PersonHasFacultyPredicate.java

```diff
@@ -0,0 +1,44 @@
+package staffconnect.model.person;
+
+import java.util.function.Predicate;
+
+import staffconnect.commons.util.ToStringBuilder;
+
+/**
+ * Tests that a {@code Person}'s {@code Faculty} matches the given faculty to filter for.
+ */
+public class PersonHasFacultyPredicate implements Predicate<Person> {
+    private final Faculty faculty;
+
+    public PersonHasFacultyPredicate(Faculty faculty) {
+        this.faculty = faculty;
+    }
+
+    @Override
+    public boolean test(Person person) {
+        if (faculty == null) {
+            return true;
+        }
+        return person.getFaculty().value.getFacultyName().equalsIgnoreCase(faculty.value.getFacultyName());
```

Comment:
Can consider using `person.getFaculty().toString().equalsIgnoreCase(faculty.toString());` instead for consistency

Reply:
will take note!